oak_functions/loader/src/server.rs

```diff
@@ -248,6 +253,43 @@ impl WasmState {
         Ok(())
     }
 
+    pub fn channel_read(
+        &mut self,
+        channel_handle: AbiChannelHandle,
+        dest_ptr_ptr: AbiPointer,
+        dest_len_ptr: AbiPointer,
+    ) -> Result<(), ChannelStatus> {
+        // Read message from channel at channel_handle.
+        let channel_handle =
+            ChannelHandle::from_i32(channel_handle).ok_or(ChannelStatus::ChannelHandleInvalid)?;
+        let endpoint = self
+            .channel_switchboard
+            .get_mut(&channel_handle)
+            .ok_or(ChannelStatus::ChannelHandleInvalid)?;
+        let receiver = &mut endpoint.receiver;
+        let message = receiver.try_recv().map_err(|e| match e {
+            TryRecvError::Empty => ChannelStatus::ChannelEmpty,
+            TryRecvError::Disconnected => ChannelStatus::ChannelEndpointDisconnected,
+        })?;
+
+        // Write message to memory of wasm module.
+        let dest_ptr = self.alloc(message.len() as u32);
+
+        let from_oak_status = |e| match e {
+            OakStatus::ErrInvalidArgs => ChannelStatus::ChannelInvalidArgs,
+            _ => ChannelStatus::Unspecified,
+        };
```

Comment:
This may be a nested function, but also it would be even nicer to define `From<OakStatus> for ChannelStatus` or something like that, so that the `?` operator does this conversion automatically.

Reply:
Done.

Reply:
Neat! Thank you! :)